### ISSUE TITLE
Zoom to bounding box after selecting filter

### ISSUE THREAD

**Issue by SRGDamia1** [ODM2/ODM2DataSharingPortal#253]
When browsing sites, it would be very cool to automatically zoom to the extent of sites remaining selected after a filter is applied.

**Comment by SRGDamia1** [comment]
The auto-zoom feature on the sandbox looks good so far.

**Comment by horsburgh** [comment]
@jcaraballo17 and @Maurier - I Just tested this and it appears to work, but when I click on the "Clear" button or deselect all of the facets after selecting them, it zooms the map to a view that shows South America and Antarctica. Not super useful. The view when the "Clear" button is selected or when all of the browser facets are actively deselected (the user unchecks them) should be the default map extent when the application loads.

**Comment by horsburgh** [comment]
The behavior of this is really buggy at the moment on the sandbox. It is zooming me into weird locations when I make different selections. For example:

* Select University of St. Thomas as the Organization and Site Type of Stream and it zooms me in out the ocean somewhere in the middle of nowhere.

I think there needs to be a default zoom level if you select a set of browsing facets for which there are no data. Either that, or you need to collapse categories based on selections in other categories like is done in TSA.

**Comment by horsburgh** [comment]
Also - Organizations are not in alphabetical order in the browsing facet. This needs to be fixed.

**Comment by Maurier** [comment]
> when I click on the "Clear" button or deselect all of the facets after selecting them, it zooms the map to a view that shows South America and Antarctica. Not super useful.

This is a bug caused by a conflict between the extent boundaries and the minimum zoom allowed in the settings. 

In order to solve it I had to allow a wider zoom which will cause blank areas to show above and below the map for such wide zoom levels.

![image](https://user-images.githubusercontent.com/2448568/41193501-2c15c092-6bca-11e8-9082-a66748f72b29.png)

> I think there needs to be a default zoom level if you select a set of browsing facets for which there are no data.

For this instance I think it's more intuitive to not issue a re-zoom since no new markers were added to the extent. The map just stays put. 

A window showing the number of results has been added to the bottom of the map to help with any confusion.

![image](https://user-images.githubusercontent.com/2448568/41194472-acc42e28-6bd8-11e8-948d-8f8742dfae1c.png)

> Also - Organizations are not in alphabetical order in the browsing facet. This needs to be fixed.

For consistency all filter items will be sorted in alphabetical order.
